### PR TITLE
Utils/WP-Exporter: New - Add ability to filter posts by meta query. ED-2101

### PR DESCRIPTION
**Pull request by iNewLegend**
Depends on https://github.com/elementor/elementor/pull/13856

### CODE REVIEW

File: core/utils/wp-exporter.php [hunk 1]
```diff
@@ -0,0 +1,749 @@
...
+	private static $default_args = [
+		'content' => 'all',
+		'author' => false,
+		'category' => false,
+		'start_date' => false,
+		'end_date' => false,
+		'status' => false,
+		'offset' => 0,
+		'limit' => -1,
+		'with_meta_key' => '', // If specified `meta_key` then will include all post(s) that have this meta_key.
```
Comment (matipojo) [inline core/utils/wp-exporter.php:33]:
```suggestion
		'meta_key' => '', // If specified `meta_key` then will include all post(s) that have this meta_key.
```

File: core/utils/wp-exporter.php [hunk 2]
```diff
@@ -0,0 +1,749 @@
...
+		if ( ! empty( $this->args['with_meta_key'] ) ) {
+			if ( $join ) {
+				$join .= ' ';
+			}
+
+			if ( $where ) {
+				$where .= ' AND ';
+			}
+
+			$join .= "INNER JOIN {$this->wpdb->postmeta} ON ({$this->wpdb->posts}.ID = {$this->wpdb->postmeta}.post_id)";
```
Comment (matipojo) [inline core/utils/wp-exporter.php:110]:
```suggestion
			$join .= "LEFT JOIN {$this->wpdb->postmeta} ON ({$this->wpdb->posts}.ID = {$this->wpdb->postmeta}.post_id)";
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/utils/wp-exporter.php (hunk 1 above)
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

File: core/utils/wp-exporter.php (hunk 2 above)
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

**Review by matipojo — APPROVED** [review]

**Comment by iNewLegend** [comment]
Handled in https://github.com/elementor/elementor/pull/13856

**Pull request closed without merging**